pipeline/psth.py

```diff
@@ -440,69 +440,33 @@ class UnitSelectivity(dj.Computed):
     -> Selectivity.proj(unit_selectivity='selectivity')
     """
 
-    key_source = ephys.Unit & PeriodSelectivity
+    # Unit Selectivity is computed only for units
+    # that has PeriodSelectivity computed for "sample" and "delay" and "response"
+    key_source = (ephys.Unit
+                  & (PeriodSelectivity & 'period = "sample"')
+                  & (PeriodSelectivity & 'period = "delay"')
+                  & (PeriodSelectivity & 'period = "response"'))
 
     def make(self, key):
         '''
         calculate 'global' selectivity for a unit -
         '''
         log.debug('UnitSelectivity.make(): key: {}'.format(key))
 
-        # verify insertion location is present,
-        egpos = None
-        try:
-            egpos = (ephys.ProbeInsertion.InsertionLocation
-                     * experiment.BrainLocation & key).fetch1()
-        except dj.DataJointError as e:
-            if 'exactly one tuple' in repr(e):
-                log.error('... Insertion Location missing. skipping')
-                return
-
         # fetch region selectivity,
-        sel = (PeriodSelectivity & key
-               & "p_value <= '{}'".format(PeriodSelectivity.alpha)).fetch(
-                   as_dict=True)
+        sels = (PeriodSelectivity & key).fetch('period_selectivity')
 
-        if not sel:
+        assert len(sels) == 3  # one more sanity check, making sure we have all 3 periods
```

Comment:
Not a good check - will fail if we ever extend PeriodSelectivity to contain other periods.

Reply:
Then we will need to change the `key_source` here as well
This check is actually redundant already because of the `key_source`, I'll remove